src/charm.py

```diff
@@ -303,6 +310,13 @@ def _on_certificate_expiring(self, event: CertificateExpiringEvent) -> None:
         )
         self._stored.csr = new_csr.decode()
 
+    def _on_cert_transfer_relation_joined(self, event: RelationJoinedEvent):
+        if self.certificates.cert:
+            certificate = self.certificates.cert
+            ca = self.certificates._ca_cert
+            chain = self.certificates._chain
+            self.mutual_tls.set_certificate(certificate=certificate, ca=ca, chain=chain)
```

Comment:
Is this stored in unit data?
Mind confirming scaling the app works?

Reply:
it is stored in unit data, not sure i see why it wouldn't work with scaling, but will give it a go as soon as I find a charm to integrate this with